`bee-autopeering` needs `rand`s `std` and `std_rng`

## Bug description

The `bee-autopeering` package cannot be built without the `rand/std` and `rand/std_rng` (default) features.

```
error[E0432]: unresolved import `rand::thread_rng`
  --> bee-network\bee-autopeering\src\discovery\query.rs:16:12
   |
16 | use rand::{thread_rng, Rng as _};
   |            ^^^^^^^^^^ no `thread_rng` in the root

error[E0425]: cannot find function `thread_rng` in crate `rand`
   --> bee-network\bee-autopeering\src\discovery\manager.rs:474:35
    |
474 |                 let index = rand::thread_rng().gen_range(0..replacements.read().len());
    |                                   ^^^^^^^^^^ not found in `rand`

error[E0425]: cannot find function `thread_rng` in crate `rand`
   --> bee-network\bee-autopeering\src\discovery\manager.rs:977:29
    |
977 |         let mut rng = rand::thread_rng();
    |                             ^^^^^^^^^^ not found in `rand`
```

This is not a problem when building the workspace because `bee-tangle` uses these features.

## Rust version

Which version of Rust are you running?

- Rust version: `rustc 1.60.0-nightly (498eeb72f 2022-01-31)`

## Bee version

Which version of Bee are you using?

- Bee version: `shimmer_develop`

## Steps To reproduce the bug

Explain how the maintainer can reproduce the bug.

1. `cargo build -p bee-autopeering`
2. ???
3. Profit

--- a/bee-network/bee-autopeering/Cargo.toml
+++ b/bee-network/bee-autopeering/Cargo.toml
@@ -26,7 +26,7 @@ num-derive = { version = "0.3.3", default-features = false  }
 num-traits = { version = "0.2.14" , default-features = false }
 priority-queue = { version = "1.2.0" , default-features = false }
 prost = { version = "0.8", default-features = false, features = [ "std" ] }
-rand = { version = "0.8.4" , default-features = false }
+rand = { version = "0.8.4" , default-features = false, features = [ "std", "std_rng" ] }
 ring = { version = "0.16.20" , default-features = false }
 rocksdb = { version = "0.17.0", default-features = false, optional = true }
 serde = { version = "1.0", default-features = false , features = [ "derive" ] }

